TFNetwork.py

```diff
@@ -1120,7 +1120,8 @@ def get_rec_step_info(self, must_exist=True):
     """
     from TFNetworkRecLayer import RecStepInfoLayer, _SubnetworkRecCell
     rec_layer = self.get_rec_parent_layer()
-    if not rec_layer:
+    # the second condition is true if all layers have been optimized out of the rec layer
+    if not rec_layer or len(rec_layer.cell.layers_in_loop) == 0:
```

Comment:
I don't understand.
If `rec_layer` is None, the code below will lead to an exception (any code below which access `rec_layer`).
Also, `must_exist` specifically says that it must exists. So if it does not exist, it should throw an exception (that is what `must_exists` is for).

Reply:
If `rec_layer` is None, the first condition is true. Then, it is ensured that `must_exist` is false, otherwise an exception is thrown as expected. If `must_exist` is false, None is returned, indicating that there is no recurrent step information. This behavior is used by `have_rec_step_info()`.

If `rec_layer` is not None, but all contained layers have been optimized out of the loop, my modification ensures that no recurrent step information is returned. This is necessary, because it was never created (see https://github.com/rwth-i6/returnn/blob/master/TFNetworkRecLayer.py#L1580).

If it is not None and contains internal layers, then the step information is returned as expected.